### PR TITLE
Use nvCOMP for Snappy compression/decompression

### PR DESCRIPTION
**Pull request by vuule**
- Use nvCOMP for Snappy compression/decompression by default.
- Change the environment variable naming to be consistent with GDS policy.
- Add detailed description of the behavior to the docs.

### CODE REVIEW

File: docs/cudf/source/basics/io-nvcomp-integration.rst [hunk 1]
```diff
@@ -0,0 +1,26 @@
+nvCOMP Integration
+=============================
+
+Some types of compression/decompression can be performed using either `nvCOMP library <https://github.com/NVIDIA/nvcomp>`_ or the internal implementation. 
+
+Which implementation is used by default depends on the data format and the compression type. Behavior can be influenced through environment variable ``LIBCUDF_NVCOMP_POLICY``.
+
+There are three special values for the environment variable:
+
+- "STABLE": only the nvCOMP usage that has been thoroughly tested in enabled.
```
Comment (JohnZed) [inline docs/cudf/source/basics/io-nvcomp-integration.rst:10]:
```suggestion
- "STABLE": only enable the nvCOMP readers that have been deemed stable for production use. 
```

File: docs/cudf/source/basics/io-nvcomp-integration.rst [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+- "EXPERIMENTAL": All available uses of nvCOMP are enabled, regardless of the level of prior testing.
```
Comment (JohnZed) [inline docs/cudf/source/basics/io-nvcomp-integration.rst:11]:
```suggestion
- "EXPERIMENTAL": All available uses of nvCOMP are enabled, including new, experimental combinations.
```

**Comment by vuule** [comment]
rerun tests

**Comment by vuule** [comment]
rerun tests

File: cpp/src/io/utilities/config_utils.hpp [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+inline bool is_stable_integration_enabled() { return get_env_policy() == usage_policy::STABLE; }
+
+/**
+ * @brief Returns true if experimental nvCOMP use is enabled.
+ *
+ * Note: When experimental use is enabled, stable use is enabled as well
+ */
+inline bool is_experimental_integration_enabled()
+{
+  return is_stable_integration_enabled() or get_env_policy() == usage_policy::EXPERIMENTAL;
```
Comment (elstehle) [inline cpp/src/io/utilities/config_utils.hpp:65]: `is_stable_integration_enabled` is currently not used, so I can only infer its semantics. But I'd assume, that `stable` should be a subset of `experimental` not the other way round? E.g., let's assume `brotli` would be experimental for nvcomp, then:
```
if (nvcomp_integration::is_experimental_integration_enabled()) {
    brotli_decompress(...);
}else{
    // libcudf code path
}
```
As I understand the conditional above, if the user has chosen `STABLE`, we'd take the code path for experimental nvcomp debrotli, even though the user actually only wanted to use the stable ones?

**Review by elstehle — COMMENTED** [review]
Overall looks good. I just think there's a mismatch between the documented values and the values actually used in the implementation.

**Review by rgsl888prabhu — COMMENTED** [review]
Looks good, just need to answer above queries.

**Review by rgsl888prabhu — APPROVED** [review]

**Review by elstehle — APPROVED** [review]

**Comment by vuule** [comment]
rerun tests

**Comment by vuule** [comment]
@gpucibot merge

**Pull request merged**